### PR TITLE
Add new `registry update` command

### PR DESCRIPTION
**Pull request by shomron**
Part of #237.

Currently supports GitHub repositories only. Helm / Fs support to be added in subsequent PR.

### Outstanding issues

- [ ] Helm repository support
- [ ] Filesystem repository support
- [ ] Update to specific version, rather than latest matching
- [ ] Tests for `TransactionWriter`
- [ ] Correct update behavior for GitHub repositories lacking a `GitVersion` section in `app.yaml`
- [ ] Ensure correct consumption of registry caches as part of #603 
- [ ] Followup PR for package update

Signed-off-by: Oren Shomron <shomron@gmail.com>

### CODE REVIEW

**Comment by shomron** [comment]
I'd love some scrutiny around error messages, wrapping semantics.

File: pkg/actions/registry_update.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+		}
+		newGitVersion.CommitSHA = newVersion
+		newRS.GitVersion = &newGitVersion
+
+		if err := a.UpdateRegistry(&newRS); err != nil {
+			return "", errors.Wrapf(err, "error updating app registry pointer: %v", rs.Name)
+		}
+	} else {
+		log.Debugf("[update] registry %v version unchanged: %v", rs.Name, newVersion)
+		// TODO where does helm store its versions?
```
Comment (bryanl) [inline pkg/actions/registry_update.go]: No where yet! Wanted to see what this change did first.

File: pkg/app/mocks/generate.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mocks
+
+//go:generate mockery -dir .. -output . -name=App
```
Comment (bryanl) [inline pkg/app/mocks/generate.go]: I like this pattern.

**Comment by bryanl** [comment]
@shomron 

Can you update to a previous SHA1?

```
 I  /p/t/mule4  ks registry update incubator --version=ac9c25c4823037d6e86b0219a3a49b0b7d57ab57 -v             400ms  Mon Jun 11 15:41:34 2018
DEBUG setting log verbosity                         verbosity-level=1
DEBUG loading application configuration from /private/tmp/mule4
DEBUG Updating registries: [incubator]

DEBUG updating registry incubator
DEBUG [github.Update] trying to update registry ksonnet/parts@master
DEBUG [github.Update] current version: 40285d8a14f1ac5787e405e1023cf0c07f6aa28c
DEBUG github: fetching SHA1 for ksonnet/parts@master
DEBUG [fetchRemoteSpec] fetching ksonnet/parts/incubator/registry.yaml@40285d8a14f1ac5787e405e1023cf0c07f6aa28c
DEBUG github: fetching contents for ksonnet/parts/incubator/registry.yaml@40285d8a14f1ac5787e405e1023cf0c07f6aa28c
DEBUG [update] registry incubator version unchanged: 40285d8a14f1ac5787e405e1023cf0c07f6aa28c
```

**Comment by bryanl** [comment]
AS for logging, we should start using logrus in a better fashion. 

Instead of doing `logrus.Debug("[github.Update]: whatever")`, why aren't we doing `logrus.WithField("action", "github.Update").Debug("whatever")`. This would be helpful if you are using ksonnet as part of an automated pipeline because logs would be generated in JSON format and you could actually search for logs with `action=github.Update`.

**Comment by shomron** [comment]
> Can you update to a previous SHA1?

Not quite yet :) It's in the outstanding issues list above.

File: pkg/app/mocks/generate.go (hunk 2 above)
Comment (shomron) [reply pkg/app/mocks/generate.go]: Thanks!

**Pull request merged**